src/libutil/logging.cc

```diff
@@ -162,17 +210,67 @@ struct JSONLogger : Logger
         nlohmann::json json;
         json["action"] = "result";
         json["id"] = act;
-        json["type"] = type;
+        json["type"] = jsonResultType(type);
         addFields(json, fields);
         write(json);
     }
 };
 
+/**
+ * A json logger intended for external consumption
+ * (contrary to 'JSONLogger' which is an internal thing
+ */
+struct ExternalJSONLogger : JSONLogger
+{
+    ExternalJSONLogger(Logger & prevLogger) : JSONLogger(prevLogger) { }
+
+    nlohmann::json jsonActivityType(ActivityType type) override
+    {
+        switch (type) {
```

Comment:
I think it's more maintainable to log the activity type and result type as a numerical value.

Reply:
It's an annoying tradeoff to make because magic numbers are definitely not user-friendly, but I agree these custom-print functions will probably cause some issue at some point (unless there's a way to auto-generate it?).

I've removed the custom json logger in 1e893d4b94981dae1aa1a0f69ae8864c7a3c47ae, so we're back to numerical values.